File: examples/governance/operations_domain/lib/operations_domain/deployment/commands/deploy_model.rb
module OperationsDomain
  class Deployment
    module Commands
      class DeployModel
        include Hecks::Command
        emits "DeployedModel"

        attr_reader :deployment_id

        def initialize(deployment_id: nil)
          @deployment_id = deployment_id
        end

        def call
          existing = repository.find(deployment_id)
          if existing
            unless existing.status == "planned"
              raise Hecks::Error, "Cannot DeployModel: status must be 'planned', got '#{existing.status}'"
            end
            Deployment.new(
              id: existing.id,
              model_id: existing.model_id,
              environment: existing.environment,
              endpoint: existing.endpoint,
              purpose: existing.purpose,
              audience: existing.audience,
              decommissioned_at: existing.decommissioned_at,
              deployed_at: Time.now.to_s,
              status: "deployed"
            )
          else
            raise Hecks::Error, "Deployment not found: #{deployment_id}"
          end
        end
      end
    end
  end
end

File: examples/governance_static_go/domain/deploy_model.go
package domain

import (
	"time"
	"fmt"
)

type DeployModel struct {
	DeploymentId string `json:"deployment_id"`
}

func (c DeployModel) CommandName() string { return "DeployModel" }

func (c DeployModel) Execute(repo DeploymentRepository) (*Deployment, *DeployedModel, error) {
	existing, err := repo.Find(c.DeploymentId)
	if err != nil {
		return nil, nil, err
	}
	if existing == nil {
		return nil, nil, fmt.Errorf("Deployment not found: %s", c.DeploymentId)
	}
	existing.UpdatedAt = time.Now()
	if err := existing.Validate(); err != nil {
		return nil, nil, err
	}
	if err := repo.Save(existing); err != nil {
		return nil, nil, err
	}
	event := DeployedModel{
		AggregateID: existing.ID,
		DeploymentId: c.DeploymentId,
		OccurredAt: time.Now(),
	}
	return existing, &event, nil
}
